src/Renci.SshNet/Channels/Channel.cs

```diff
@@ -715,8 +714,14 @@ private void OnChannelRequest(object sender, MessageEventArgs<ChannelRequestMess
                     }
                     else
                     {
-                        // TODO: we should also send a SSH_MSG_CHANNEL_FAILURE message
-                        throw new NotSupportedException(string.Format(CultureInfo.CurrentCulture, "Request '{0}' is not supported.", e.Message.RequestName));
+                        var unknownRequestInfo = new UnknownRequestInfo(e.Message.RequestName);
+                        unknownRequestInfo.Load(e.Message.RequestData);
+
+                        if (unknownRequestInfo.WantReply)
+                        {
+                            var reply = new ChannelFailureMessage(LocalChannelNumber);
```

Comment:
I think this should be:

```suggestion
                            var reply = new ChannelFailureMessage(RemoteChannelNumber);
```

Unfortunately, it seems like these lines are wrong in SshCommand:

https://github.com/sshnet/SSH.NET/blob/db3d7e8d03d0a852cd7d16e1dee95a72939de317/src/Renci.SshNet/SshCommand.cs#L461-L481

Reply:
Fixed it in Channel.cs . Should I fix SshCommand.cs as well? (Seems unrelated to this issue to me).

Reply:
Thanks, if it is a one-line fix, it would be good to fix it here. If it is more effort, then you can leave it.

Reply:
Pushed it (had to extend IChannel interface with RemoteChannelNumber).